dm/syncer/dml_worker.go

```diff
@@ -196,17 +196,24 @@ func (w *DMLWorker) executeJobs(queueID int, jobCh chan *job) {
 // executeBatchJobs execute jobs with batch size.
 func (w *DMLWorker) executeBatchJobs(queueID int, jobs []*job) {
 	var (
-		affect int
-		db     = w.toDBConns[queueID]
-		err    error
-		dmls   = make([]*DML, 0, len(jobs))
+		affect  int
+		queries []string
+		args    [][]interface{}
+		db      = w.toDBConns[queueID]
+		err     error
+		dmls    = make([]*DML, 0, len(jobs))
 	)
 
 	defer func() {
 		if err == nil {
 			w.successFunc(queueID, len(dmls), jobs)
 		} else {
-			w.fatalFunc(jobs[affect], err)
+			if len(queries) == len(jobs) {
+				w.fatalFunc(jobs[affect], err)
+			} else {
+				w.logger.Warn("length of queries not equals length of jobs, cannot determine which job failed, use first one instead", zap.Int("queries", len(queries)), zap.Int("jobs", len(jobs)))
+				w.fatalFunc(jobs[0], err)
```

Comment:
Since we can determine the exact job that trigger this failure, we should add a log about the first and last job's binlog location. Or maybe we can create a new job with the start location of the first job and end location of the last job.

Reply:
May not, if we want to inject DDL before the error DDL, the first job is better then all jobs.